### PR TITLE
add auto-generated configuration code

### PR DESCRIPTION
**Pull request by codeboten**
This is generated from the opentelemetry-configuration repository using gojsonschema. Updated makefile to support this w/ `make genjsonschema`

Follow up to https://github.com/open-telemetry/opentelemetry-collector/pull/7678

### CODE REVIEW

File: service/telemetry/generated_config.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// Code generated by github.com/atombender/go-jsonschema, DO NOT EDIT.
+
+package telemetry
+
+type Attributes struct {
+	// ServiceName corresponds to the JSON schema field "service.name".
+	ServiceName *string `json:"service.name,omitempty" yaml:"service.name,omitempty" mapstructure:"service.name,omitempty"`
```
Comment (bogdandrutu) [inline service/telemetry/generated_config.go:7]: Can we configure the tool to only set the mapstructure tags?

Comment (codeboten) [reply service/telemetry/generated_config.go:7]: The tool doesn't support this at the moment: https://github.com/omissis/go-jsonschema/blob/172061399cba4a93f95ac5cf3b3935daa147e966/pkg/generator/generate.go#L807-L811

Comment (codeboten) [reply service/telemetry/generated_config.go:7]: Added a feature request here: https://github.com/omissis/go-jsonschema/issues/79

**Review by TylerHelmuth — APPROVED** [review]

File: service/telemetry/generated_config.go (hunk 1 above)
Comment (omissis) [reply service/telemetry/generated_config.go:12]: hi @bogdandrutu , go-jsonschema mantainer here :) I was wondering if the extra tags are causing any trouble, or if it's just for "cosmetic" reasons, eg: for maintaining a consistent code style or something. Thanks!

Comment (bogdandrutu) [reply service/telemetry/generated_config.go:12]: It is extra public API for us that we don't want to expose. Once exposed others may rely on them, and I don't want to maintain these.

Comment (codeboten) [reply service/telemetry/generated_config.go]: for now i've updated gojsonschema to use the updated version in my fork.

File: internal/tools/go.sum [hunk 2]
```diff
@@ -732,8 +738,8 @@ go.opentelemetry.io/build-tools/semconvgen v0.8.0/go.mod h1:h0FPWeTZw4sfjNhfFdN3
 go.tmz.dev/musttag v0.7.0 h1:QfytzjTWGXZmChoX0L++7uQN+yRCPfyFm+whsM+lfGc=
 go.tmz.dev/musttag v0.7.0/go.mod h1:oTFPvgOkJmp5kYL02S8+jrH0eLrBIl57rzWeA26zDEM=
 go.uber.org/atomic v1.4.0/go.mod h1:gD2HeocX3+yG+ygLZcrzQJaqmWj9AIm7n08wl/qW/PE=
-go.uber.org/atomic v1.10.0 h1:9qC72Qh0+3MqyJbAn8YU5xVq1frD8bn3JtD2oXtafVQ=
-go.uber.org/atomic v1.10.0/go.mod h1:LUxbIzbOniOlMKjJjyPfpl4v+PKK2cNJn91OQbhoJI0=
+go.uber.org/atomic v1.9.0 h1:ECmE8Bn/WFTYwEW/bpKD3M8VtR/zQVbavAoalC1PYyE=
+go.uber.org/atomic v1.9.0/go.mod h1:fEN4uk6kAWBTFdckzkM89CLk9XfWZrxpCo0nPH17wJc=
```
Comment (bogdandrutu) [inline internal/tools/go.sum:742]: Sure about these changes?

File: internal/tools/go.sum [hunk 3]
```diff
@@ -534,31 +541,27 @@ github.com/prometheus/client_golang v0.9.3/go.mod h1:/TN21ttK/J9q6uSwhBd54HahCDf
...
+github.com/prometheus/common v0.32.1 h1:hWIdL3N2HoUx3B8j3YN9mWor0qhY/NlEKZEaXxuIRh4=
 github.com/prometheus/common v0.32.1/go.mod h1:vu+V0TpY+O6vW9J44gczi3Ap/oXXR10b+M/gUGO4Hls=
-github.com/prometheus/common v0.44.0 h1:+5BrQJwiBB9xsMygAB3TNvpQKOwlkc25LbISbrdOOfY=
-github.com/prometheus/common v0.44.0/go.mod h1:ofAIvZbQ1e/nugmZGz4/qCb9Ap1VoSTIO7x0VV9VvuY=
 github.com/prometheus/procfs v0.0.0-20181005140218-185b4288413d/go.mod h1:c3At6R/oaqEKCNdg8wHV1ftS6bRYblBhIjjI8uT2IGk=
 github.com/prometheus/procfs v0.0.0-20190507164030-5867b95ac084/go.mod h1:TjEm7ze935MbeOT/UhFTIMYKhuLP4wbCsTZCD3I8kEA=
 github.com/prometheus/procfs v0.0.2/go.mod h1:TjEm7ze935MbeOT/UhFTIMYKhuLP4wbCsTZCD3I8kEA=
 github.com/prometheus/procfs v0.1.3/go.mod h1:lV6e/gmhEcM9IjHGsFOCxxuZ+z1YqCvr4OA4YeYWdaU=
 github.com/prometheus/procfs v0.6.0/go.mod h1:cz+aTbrPOrUb4q7XlbU9ygM+/jj0fzG6c1xBZuNvfVA=
+github.com/prometheus/procfs v0.7.3 h1:4jVXhlkAyzOScmCkXBTOLRLTz8EeU+eyjrwB/EPq0VU=
```
Comment (bogdandrutu) [inline internal/tools/go.sum:563]: downgrade?

File: internal/tools/go.sum [hunk 4]
```diff
@@ -534,31 +541,27 @@ github.com/prometheus/client_golang v0.9.3/go.mod h1:/TN21ttK/J9q6uSwhBd54HahCDf
...
+github.com/prometheus/client_model v0.2.0 h1:uq5h0d+GuxiXLJLNABMgp2qUWDPiLvgCzz2dUR+/W/M=
 github.com/prometheus/client_model v0.2.0/go.mod h1:xMI15A0UPsDsEKsMN9yxemIoYk6Tm2C1GtYGdfGttqA=
-github.com/prometheus/client_model v0.4.0 h1:5lQXD3cAg1OXBf4Wq03gTrXHeaV0TQvGfUooCfx1yqY=
-github.com/prometheus/client_model v0.4.0/go.mod h1:oMQmHW1/JoDwqLtg57MGgP/Fb1CJEYF2imWWhWtMkYU=
 github.com/prometheus/common v0.0.0-20181113130724-41aa239b4cce/go.mod h1:daVV7qP5qjZbuso7PdcryaAu0sAZbrN9i7WWcTMWvro=
 github.com/prometheus/common v0.4.0/go.mod h1:TNfzLD0ON7rHzMJeJkieUDPYmFC7Snx/y86RQel1bk4=
 github.com/prometheus/common v0.4.1/go.mod h1:TNfzLD0ON7rHzMJeJkieUDPYmFC7Snx/y86RQel1bk4=
 github.com/prometheus/common v0.10.0/go.mod h1:Tlit/dnDKsSWFlCLTWaA1cyBgKHSMdTB80sz/V91rCo=
 github.com/prometheus/common v0.26.0/go.mod h1:M7rCNAaPfAosfx8veZJCuw84e35h3Cfd9VFqTh1DIvc=
+github.com/prometheus/common v0.32.1 h1:hWIdL3N2HoUx3B8j3YN9mWor0qhY/NlEKZEaXxuIRh4=
```
Comment (bogdandrutu) [inline internal/tools/go.sum:556]: downgrade?

**Review by bogdandrutu — APPROVED** [review]
Lots of downgrades in the go.sum is that expected?

File: service/telemetry/generated_config.go (hunk 1 above)
Comment (omissis) [reply service/telemetry/generated_config.go]: hi @codeboten sorry for the delay! I reviewed and merged the PR, and release a new v0.12.0 as well :)

**Pull request merged**